hazelcast/src/main/java/com/hazelcast/config/IndexConfig.java

```diff
@@ -218,6 +221,27 @@ public IndexConfig setBitmapIndexOptions(BitmapIndexOptions bitmapIndexOptions)
         return this;
     }
 
+    /**
+     * Provides access to index options specific to B-Tree indexes.
+     *
+     * @return the b-tree index configuration associated with this config.
+     */
+    public BTreeIndexConfig getBTreeIndexConfig() {
+        if (bTreeIndexConfig == null) {
```

Comment:
I didn't get why this field can be null. Why don't we use the `new BTreeIndexConfig()` as the default value and not allow user to set it to null?

Reply:
I see. I couldn't think of a scenario where having this null by default would help other than maybe trying to carry fewer bytes over the network when this index config is serialized unless the user sets it to something non-default.

Reply:
We've discussed this on daily, there is no hard convention, so I will follow your comment - it's simpler :)